gitlab-pipeline/stage/test.yml

```diff
@@ -731,7 +743,41 @@ test_full_integration_enterprise:
     reports:
       junit: results_full_integration.xml
 
-test_full_integration_open_source:
-  extends: test_full_integration_enterprise
+
+test:integration:open_source:build_client:
+  extends: .integration_setup_template
+  only:
+    variables:
+      - ( $RUN_INTEGRATION_TESTS == "true" && $BUILD_CLIENT == "true" )
+  needs:
+    - init_workspace
+    - build_servers
+    - build_client
+    - build_client_docker
+    - build_mender-artifact
+
+
+test:integration:enterprise:build_client:
+  extends: test:integration:open_source:build_client
   variables:
-    TEST_SUITE: "open"
+    TEST_SUITE: "enterprise"
+
+
+test:integration:open_source:without_build_client:
```

Comment:
Minor suggestion, what about `test:integration:open_source:prebuilt_client` ?

Reply:
oh, yes. I was playing with `fresh_client` and the like, but didn't find anything I liked. Are you mastering the most difficult skill in CS completely these days? :p